liquibase-service/src/main/resources/db/odse/routines/021-sp_interview_event-001.sql

```diff
@@ -0,0 +1,1291 @@
+
+        INSERT INTO [rdb_modern].[dbo].[job_flow_log]
+        (batch_id, [Dataflow_Name], [package_Name], [Status_Type], [step_number], [step_name], [row_count])
+        VALUES (@batch_id, 'Interview PRE-Processing Event', 'nrt_interview', 'START', @Proc_Step_no, @Proc_Step_Name,
+                @RowCount_no);
+
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN
+            TRANSACTION;
+        SET
+            @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET
+            @PROC_STEP_NAME = 'GENERATING #TEXT_FINAL';
+
+        SELECT NBS_ANSWER_UID,
+               RDB_COLUMN_NM,
+               ANSWER_TXT,
+               INTERVIEW_UID,
+               NBS_QUESTION_UID
+        INTO #TEXT_FINAL
+        FROM (SELECT DISTINCT NBS_ANSWER_UID,
+                              CODE_SET_GROUP_ID,
+                              RDB_COLUMN_NM,
+                              ANSWER_TXT,
+                              ACT_UID AS INTERVIEW_UID,
+                              RECORD_STATUS_CD,
+                              NBS_QUESTION_UID,
+                              data_type,
+                              rdb_table_nm,
+                              answer_group_seq_nbr
+              from dbo.v_rdb_ui_metadata_answers
+              WHERE ACT_UID IN (SELECT value FROM STRING_SPLIT(@ix_uids, ','))) as metadata
+                 INNER JOIN NBS_SRTE.DBO.CODE_VALUE_GENERAL CVG
+                            ON UPPER(CVG.CODE) = UPPER(DATA_TYPE)
+        WHERE CVG.CODE_SET_NM = 'NBS_DATA_TYPE'
+          AND CODE = 'TEXT'
+          and rdb_table_nm = 'D_INTERVIEW'
+          AND ANSWER_GROUP_SEQ_NBR IS NULL
+
+
+        if
+            @debug = 'true'
+            select @Proc_Step_Name as step, *
+            from #TEXT_FINAL;
+
+        SELECT @RowCount_no = @@ROWCOUNT;
+
+        INSERT INTO [rdb_modern].[dbo].[job_flow_log]
+        (batch_id, [Dataflow_Name], [package_Name], [Status_Type], [step_number], [step_name], [row_count])
+        VALUES (@batch_id, 'Interview PRE-Processing Event', 'nrt_interview', 'START', @Proc_Step_no, @Proc_Step_Name,
+                @RowCount_no);
+
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN
+            TRANSACTION;
+        SET
+            @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET
+            @PROC_STEP_NAME = 'GENERATING #NUMERIC_BASE_DATA';
+
+        SELECT NBS_ANSWER_UID,
+               metadata.CODE_SET_GROUP_ID,
+               RDB_COLUMN_NM,
+               ANSWER_TXT,
+               PA.ACT_UID AS INTERVIEW_UID,
+               PA.RECORD_STATUS_CD,
+               metadata.NBS_QUESTION_UID
+        INTO #NUMERIC_BASE_DATA
+        FROM (SELECT DISTINCT RDB_COLUMN_NM,
+                              NBS_QUESTION_UID,
+                              CODE_SET_GROUP_ID,
+                              INVESTIGATION_FORM_CD,
+                              QUESTION_GROUP_SEQ_NBR,
+                              DATA_TYPE
+              FROM dbo.v_rdb_ui_metadata_answers
```

Comment:
Are there any 1:many relationships or do most of the joins line up 1:1 for rdb table schema-question-format-answer?

Reply:
There are 1:many relationships, so I had to throw DISTINCT in there. The original SAS has it as well

Reply:
Which column causes the 1:many relationship? If the baseline uses distinct that should be good to go 👍

Reply:
![image](https://github.com/user-attachments/assets/fc5a0232-c6a4-4ce2-a0d1-7a11907dbc8a)


This section from the view definition